### DISCUSSION TITLE
Upgrade from 1.71.1 to 1.94.1 fails

### DISCUSSION THREAD

**Discussion by rjanossy** [kimai/kimai#3280] [General]
Hi,
I installed the latest Ubuntu 22.04 LTS on my new server.
Than followed the instructions here:
https://www.kimai.org/documentation/fresh-ubuntu-20.html

The only difference was here:
git clone -b **1.71.1** --depth 1 https://github.com/kevinpapst/kimai2.git
I wanted to use the 1.71.1 because we have an older system with this version running well.

1,71.1 was up and running. Then the database was restored from the older other machine (Ubuntu 20.04 , Kimai2 1.71.1)
After restoring the DB, the 1.71.1was working well.

Later I wanted to upgrade to the latest version, the 1.94.1.
I followed these instructions step-by-step:
https://www.kimai.org/documentation/updates.html

And ended up with this error:

![kimai_error](https://user-images.githubusercontent.com/97611067/165755022-a1e2d4b2-4ad1-41d3-bd43-a818dca8aedc.jpg)

Any ideas how to solve this?

Regards,
Robert

**Comment by kevinpapst** [comment]
This story is missing some details, otherwise this would not have happened.
Anyway: simply delete the mentioned database table and re-run the update.

**Comment by rjanossy** [comment]
Hi,
I re-installed everything, starting from Ubuntu 22.04 install......
The 1.94.1 was running well, without any data of course.
I restored the data from the 1.71.1 machine.

Starting at x.x.x.x/hu/login........
![kimai_login](https://user-images.githubusercontent.com/97611067/165795136-e0364035-7293-4d37-87e5-38fabbf254f5.jpg)

After login always the error page /........./hu/timesheet 
![Error_500](https://user-images.githubusercontent.com/97611067/165796153-048c4c6b-b232-447b-a4d3-cea323b95516.jpg)

And the prod.log

[2022-04-28 17:48:50] request.INFO: Matched route "timesheet". {"route":"timesheet","route_parameters":{"_route":"timesheet","page":1,"_controller":"App\\Controller\\TimesheetController::indexAction","_locale":"hu"},"request_uri":"http:/
/x.x.x.x/hu/timesheet/","method":"GET"} []
[2022-04-28 17:48:50] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\InvalidFieldNameException: "An exception occurred while executing 'SELECT k0_.id AS id_0, k0_.name AS name_1, k0_.number AS number_2, k0_.comment AS c
omment_3, k0_.visible AS visible_4, k0_.billable AS billable_5, k0_.company AS company_6, k0_.vat_id AS vat_id_7, k0_.contact AS contact_8, k0_.address AS address_9, k0_.country AS country_10, k0_.currency AS currency_11, k0_.phone AS ph
one_12, k0_.fax AS fax_13, k0_.mobile AS mobile_14, k0_.email AS email_15, k0_.homepage AS homepage_16, k0_.timezone AS timezone_17, k0_.budget AS budget_18, k0_.time_budget AS time_budget_19, k0_.budget_type AS budget_type_20, k0_.color
 AS color_21 FROM kimai2_customers k0_ WHERE k0_.visible = ? ORDER BY k0_.name ASC' with params [true]:  SQLSTATE[42S22]: Column not found: 1054 Unknown column 'k0_.billable' in 'field list'" at /var/www/kimai2/vendor/doctrine/dbal/lib/D
octrine/DBAL/Driver/AbstractMySQLDriver.php line 79 {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 0): An exception occurred while executing 'SELECT k0_.id AS id_0, k0_.name AS name_1, k0_.number AS nu
mber_2, k0_.comment AS comment_3, k0_.visible AS visible_4, k0_.billable AS billable_5, k0_.company AS company_6, k0_.vat_id AS vat_id_7, k0_.contact AS contact_8, k0_.address AS address_9, k0_.country AS country_10, k0_.currency AS curr
ency_11, k0_.phone AS phone_12, k0_.fax AS fax_13, k0_.mobile AS mobile_14, k0_.email AS email_15, k0_.homepage AS homepage_16, k0_.timezone AS timezone_17, k0_.budget AS budget_18, k0_.time_budget AS time_budget_19, k0_.budget_type AS b
udget_type_20, k0_.color AS color_21 FROM kimai2_customers k0_ WHERE k0_.visible = ? ORDER BY k0_.name ASC' with params [true]:\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'k0_.billable' in 'field list' at /var/www/kimai2/v
endor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:79, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'k0_.billable' in 'field list' at /var/www/kimai2/vendor
/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDO/Exception.php:18, PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'k0_.billable' in 'field list' at /var/www/kimai2/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driv
er/PDOStatement.php:117)"} []

**Reply by kevinpapst** [reply]
You are doing weird stuff.
Please read again what I have written earlier. If you don't understand it clearly: ask.

Now: re-import the 1.17* backup, drop the table that is mentioned in the initial error message and then re-run the update.

**Comment by rjanossy** [comment]
Hi,THX for your time dealing with my weird stuff.
My aim is simply to set up a new server with the latest ubuntu and kimai version and mirror the existing data from my old server running 1.71.1I was not able to do this so far.

1, Dumping the data from the 1.71.1 and re-loading on the same system or on another with 1.71.1 is working. This is my general method to have backups.2, Dumping the data from the 1.71.1 and re-loading on a new machine with 1.94.1 is not working for me. You can see the results in my post from yesterday.

At this point direct upgrade from 1.71.1 to 1.94.1 is not on the table, because the second method should work.
Regards,Robert

    On Thursday, April 28, 2022, 09:59:15 PM GMT+2, Kevin Papst ***@***.***> wrote:  
 
 


You are doing weird stuff.
Please read again what I have written earlier. If you don't understand it clearly: ask.

Now: re-import the 1.17* backup, drop the table that is mentioned in the initial error message and then re-run the update.

**Comment by rjanossy** [comment]
Hmm, it seems I was wrong.....again
The only way is to do the "update" 
So back to your suggestion 'drop the table that is mentioned in the initial error message'.
It's working now.
Thank you for your patience.
Regards,
Robert

**Reply by kevinpapst** [reply]
Glad to read that
